tests/test_series.py

```diff
@@ -565,6 +565,153 @@ def test_types_groupby_methods() -> None:
     check(assert_type(s.groupby(level=0).idxmin(), pd.Series), pd.Series)
 
 
+def test_groupby_result() -> None:
+    # GH 142
+    # since there are no columns in a Series, groupby name only works
+    # with a named index, we use a MultiIndex, so we can group by more
+    # than one level and test the non-scalar case
+    multi_index = pd.MultiIndex.from_frame(
+        pd.DataFrame(
+            {"a": [0, 1, 1], "b": [0, 1, 0]},
+            # avoid ambiguous integer size on Windows
+            dtype=np.int64,
+        )
+    )
+    s = pd.Series([0, 1, 2], index=multi_index, dtype=int)
+    # workaround different default integer size on Windows, the series
+    # constructor does not work yet with IntDtypeArg, but astype does
+    s = s.astype(np.int64)
```

Comment:
I think you can avoid this by using `np.integer` in the `check` tests below instead of `np.int64`